src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -110,7 +110,7 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
             Debug.Assert((object)rewrittenType != null);
             @checked = @checked &&
                 (_inExpressionLambda && (explicitCastInCode || DistinctSpecialTypes(rewrittenOperand.Type, rewrittenType)) ||
-                NeedsChecked(rewrittenOperand.Type, rewrittenType));
+                NeedsCheckedConversion(rewrittenOperand.Type, rewrittenType, compareNumericTypes: true));
```

Comment:
>NeedsCheckedConversion(rewrittenOperand.Type, rewrittenType, compareNumericTypes: true)); [](start = 16, length = 89)

The same comment is here, I think we should simply propagate what is given as an input and let IL generation to ignore it when it is not applicable. I think this what VB does.